### ISSUE TITLE
Build fails installing on macOS 11.3.1 (Apple Silicon, M1)

### ISSUE THREAD

**Issue by lonkle** [jpochyla/psst#51]
This is the error I get when `cargo build --release` **OR** `cargo build --release`  on my M1 MacBook Air (macOS 11.3.1).
```
error: failed to run custom build command for `ep-miniaudio-sys v2.4.0 (https://github.com/jpochyla/miniaudio-rs#1ff94282)`
```

**Comment by jpochyla** [comment]
Do you have the submodules fetched? I don't have any M1 hw and have no way to test, sorry. PRs welcome!

**Comment by lonkle** [comment]
I think I can fix it, I just need to figure out how to enable `bindgen` on your build. I needed to install it I believe so:

`brew install llvm`
`cargo install bindgen`

worked for that but your build code still isn't using it. 

```
thread 'main' panicked at 'bindings for macos-aarch64 do not exist in `bindings/macos-aarch64`, please enable the `bindgen` feature instead', /Users/joel/.cargo/git/checkouts/miniaudio-rs-0268a3ab613f78e9/1ff9428/miniaudio-sys/build.rs:60:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Any suggestion on how to force it to use `bindgen`

**Comment by jpochyla** [comment]
Thanks @lonkle! Can you try adding `features = ["bindgen"]` do the `miniaudio` dependency?

**Comment by lonkle** [comment]
https://github.com/jpochyla/psst/pull/61 (you can choose to adding that "feature" conditional if you want, but `bindgen` works for all platforms it looks like).

**Comment by jpochyla** [comment]
Merged #62, thank you!

**Comment by jpochyla** [comment]
@lonkle Hi, I think we were a bit too quick with merging this. Enabling the `bindgen` feature requires `llvm` installation, and I think we can do better. Do you think it would be possible to upstream the pre-generated arm64 bindings to `miniaudio-rs`, by [running this script](https://github.com/ExPixel/miniaudio-rs/blob/master/generate-bindings.rs) and opening a PR there?

**Comment by lonkle** [comment]
@jpochyla Good point. Done! https://github.com/ExPixel/miniaudio-rs/pull/65

**Comment by jpochyla** [comment]
This is taking a bit long, let's use our fork of `miniaudio-rs`.

**Comment by jpochyla** [comment]
I've merged the PR into `jpochyla/miniaudio-rs` and included it in a71179f. Please let me know in case of further problems on M1!

**Comment by naymapl** [comment]
Hello.
I start to build fresh build on Apple Silicon and got error:

```
 Compiling psst-core v0.1.0 (/Users/naymapl/psst/psst-core)
error[E0599]: no variant or associated item named `compatible_audio_formats` found for enum `MediaFile` in the current scope
  --> psst-core/src/metadata.rs:48:31
   |
48 |         let file = MediaFile::compatible_audio_formats(preferred_bitrate)
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `MediaFile`
   |
  ::: psst-core/src/player/file.rs:35:1
   |
35 | pub enum MediaFile {
   | ------------------ variant or associated item `compatible_audio_formats` not found here

For more information about this error, try `rustc --explain E0599`.
error: could not compile `psst-core` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```
